
Upstream-Status: Inappropriate No time for upstreaming

--- a/drivers/gpu/drm/bridge/lvds-codec.c
+++ b/drivers/gpu/drm/bridge/lvds-codec.c
@@ -25,6 +25,8 @@
 	struct regulator *vcc;
 	struct gpio_desc *powerdown_gpio;
 	u32 connector_type;
+	u32 max_frequency;
+	u32 min_frequency;
 	unsigned int bus_format;
 };
 
@@ -97,7 +99,23 @@
 	return input_fmts;
 }
 
+static enum drm_mode_status lvds_codec_mode_valid(struct drm_bridge *bridge,
+					const struct drm_display_info *info,
+					const struct drm_display_mode *mode)
+{
+	struct lvds_codec *lvds_codec = to_lvds_codec(bridge);
+
+	if (mode->clock < lvds_codec->min_frequency)
+		return MODE_CLOCK_LOW;
+
+	if (mode->clock > lvds_codec->max_frequency)
+		return MODE_CLOCK_HIGH;
+
+	return MODE_OK;
+}
+
 static const struct drm_bridge_funcs funcs = {
+	.mode_valid = lvds_codec_mode_valid,
 	.attach = lvds_codec_attach,
 	.enable = lvds_codec_enable,
 	.disable = lvds_codec_disable,
@@ -188,6 +206,14 @@
 			DRM_BUS_FLAG_PIXDATA_SAMPLE_NEGEDGE;
 	}
 
+	if (!of_property_read_u32(dev->of_node, "max-frequency", &lvds_codec->max_frequency))
+		lvds_codec->max_frequency /= 1000;
+	 else
+	 	lvds_codec->max_frequency = -1;	 	
+
+	if (!of_property_read_u32(dev->of_node, "min-frequency", &lvds_codec->min_frequency))
+		lvds_codec->min_frequency /= 1000;
+
 	/*
 	 * The panel_bridge bridge is attached to the panel's of_node,
 	 * but we need a bridge attached to our of_node for our user
