
Upstream-Status: Inappropriate No time for upstreaming

--- a/drivers/net/phy/dp83867.c
+++ b/drivers/net/phy/dp83867.c
@@ -352,25 +361,30 @@
 static irqreturn_t dp83867_handle_interrupt(struct phy_device *phydev)
 {
 	int irq_status, irq_enabled;
+	int res = IRQ_NONE; 
 
-	irq_status = phy_read(phydev, MII_DP83867_ISR);
-	if (irq_status < 0) {
-		phy_error(phydev);
-		return IRQ_NONE;
-	}
+	for (;;) {
+		irq_enabled = phy_read(phydev, MII_DP83867_MICR);
+		if (irq_enabled < 0) {
+			phy_error(phydev);
+			break;
+		}
 
-	irq_enabled = phy_read(phydev, MII_DP83867_MICR);
-	if (irq_enabled < 0) {
-		phy_error(phydev);
-		return IRQ_NONE;
-	}
+		irq_status = phy_read(phydev, MII_DP83867_ISR);
+		if (irq_status < 0) {
+			phy_error(phydev);
+			break;
+		}
 
-	if (!(irq_status & irq_enabled))
-		return IRQ_NONE;
+		if (!(irq_status & irq_enabled))
+			break;
 
-	phy_trigger_machine(phydev);
+		phy_trigger_machine(phydev);
 
-	return IRQ_HANDLED;
+		res = IRQ_HANDLED;
+	}
+
+	return res;
 }
 
 static int dp83867_read_status(struct phy_device *phydev)
