--- a/drivers/net/ethernet/ti/davinci_mdio.c
+++ b/drivers/net/ethernet/ti/davinci_mdio.c
@@ -517,6 +517,11 @@
 	{ .family = "J7200", .data = &am65_mdio_soc_data },
 	{ .family = "J721E", .data = &am65_mdio_soc_data },
 	{ .family = "J721S2", .data = &am65_mdio_soc_data },
+	{ .family = "J784S4", .data = &am65_mdio_soc_data },
+	{ .family = "AM62AX", .data = &am65_mdio_soc_data },
+	{ .family = "AM62PX", .data = &am65_mdio_soc_data },
+	{ .family = "J722S", .data = &am65_mdio_soc_data },
+	{ .family = "AM62LX", .data = &am65_mdio_soc_data },
 	{ /* sentinel */ },
 };
 
