--- a/drivers/input/touchscreen/goodix.c
+++ b/drivers/input/touchscreen/goodix.c
@@ -1172,6 +1172,36 @@
 	/* Read configuration and apply touchscreen parameters */
 	goodix_read_config(ts);
 
+	int trig = irq_get_trigger_type(ts->client->irq);
+	int ctrig = goodix_irq_flags[ts->int_trigger_type];
+
+	if (trig != ctrig) {
+		int i;
+
+		dev_info(&ts->client->dev,
+			"IRQ trigger type 0x%x does not match config 0x%x, trying to fix it\n",
+			trig, ctrig);
+
+		for (i = 0; i < ARRAY_SIZE(goodix_irq_flags); i++) {
+			if (goodix_irq_flags[i] == trig) {
+				new_trigger_type = i;
+				break;
+			}
+		}
+
+		if (i < ARRAY_SIZE(goodix_irq_flags)) {
+			int error = goodix_i2c_write_u8(ts->client,
+				ts->chip->config_addr + TRIGGER_LOC,
+				(ts->config[TRIGGER_LOC] & ~0x03) | i);
+			if (error)
+				dev_err(&ts->client->dev,
+					"Failed to write new trigger type: %d\n",
+					error);
+			else
+				ts->int_trigger_type = i;
+		}
+	}
+
 	/* Try overriding touchscreen parameters via device properties */
 	touchscreen_parse_properties(ts->input_dev, true, &ts->prop);
 
