
Upstream-Status: Inappropriate No time for upstreaming

diff --git a/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi b/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
index e0d8b2a10b5f..9fd26b33da36 100644
--- a/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-am62p-j722s-common-main.dtsi
@@ -1134,14 +1134,6 @@ dphy0: phy@30110000 {
 		status = "disabled";
 	};
 
-	dphy0: phy@30110000 {
-		compatible = "cdns,dphy-rx";
-		reg = <0x00 0x30110000 0x00 0x1100>;
-		#phy-cells = <0>;
-		power-domains = <&k3_pds 185 TI_SCI_PD_EXCLUSIVE>;
-		status = "disabled";
-	};
-
 	dss0: dss@30200000 {
 		compatible = "ti,am62p51-dss";
 		reg = <0x00 0x30200000 0x00 0x1000>, /* common */
