
Upstream-Status: Inappropriate No time for upstreaming

diff --git a/drivers/gpu/imx/lcdifv3/lcdifv3-common.c b/drivers/gpu/imx/lcdifv3/lcdifv3-common.c
index 7bdb00e2ecc7..dfe95675085b 100644
--- a/drivers/gpu/imx/lcdifv3/lcdifv3-common.c
+++ b/drivers/gpu/imx/lcdifv3/lcdifv3-common.c
@@ -644,14 +644,14 @@ static int imx_lcdifv3_probe(struct platform_device *pdev)
 
 	lcdifv3->irq = platform_get_irq(pdev, 0);
 	if (lcdifv3->irq < 0) {
-		dev_err(dev, "No irq get, ret=%d\n", lcdifv3->irq);
+		dev_err_probe(dev, lcdifv3->irq, "No irq get\n");
 		return lcdifv3->irq;
 	}
 
 	lcdifv3->clk_pix = devm_clk_get(dev, "pix");
 	if (IS_ERR(lcdifv3->clk_pix)) {
 		ret = PTR_ERR(lcdifv3->clk_pix);
-		dev_err(dev, "No pix clock get: %d\n", ret);
+		dev_err_probe(dev, ret, "No pix clock get\n");
 		return ret;
 	}
 
