
Upstream-Status: Inappropriate No time for upstreaming

diff --git a/drivers/gpu/drm/imx/imx8mp-hdmi-pavi.c b/drivers/gpu/drm/imx/imx8mp-hdmi-pavi.c
index d93694d80334..95c39b7149bf 100644
--- a/drivers/gpu/drm/imx/imx8mp-hdmi-pavi.c
+++ b/drivers/gpu/drm/imx/imx8mp-hdmi-pavi.c
@@ -131,8 +131,8 @@ static int imx8mp_hdmi_pavi_probe(struct platform_device *pdev)
 
 	pavi->clk_apb = devm_clk_get(dev, NULL);
 	if (IS_ERR(pavi->clk_apb)) {
-		dev_err(dev, "No pai clock get\n");
-		return -EPROBE_DEFER;
+		dev_err_probe(dev, PTR_ERR(pavi->clk_apb), "No pai clock get\n");
+		return PTR_ERR(pavi->clk_apb);
 	}
 
 	platform_set_drvdata(pdev, pavi);
