
Upstream-Status: Inappropriate No time for upstreaming

diff --git a/drivers/md/dm-init.c b/drivers/md/dm-init.c
index c5220547ea54..915ac9e2a473 100644
--- a/drivers/md/dm-init.c
+++ b/drivers/md/dm-init.c
@@ -296,7 +296,7 @@ static int __init dm_init_init(void)
 			dev_t dev;
 
 			DMINFO("waiting for device %s ...", waitfor[i]);
-			while (early_lookup_bdev(waitfor[i], &dev))
+			while (early_lookup_bdev(waitfor[i], &dev) || !driver_probe_done())
 				fsleep(5000);
 		}
 	}
