--- a/drivers/gpu/drm/imx/dw_mipi_dsi-imx.c
+++ b/drivers/gpu/drm/imx/dw_mipi_dsi-imx.c
@@ -442,7 +442,8 @@
 
 	ret = drm_bridge_attach(&dsi->encoder, bridge, NULL, 0);
 	if (ret) {
-		DRM_DEV_ERROR(dev, "failed to attach bridge: %d\n", ret);
+		if (ret != -EPROBE_DEFER)
+			DRM_DEV_ERROR(dev, "failed to attach bridge: %d\n", ret);
 		return ret;
 	}
 
@@ -516,7 +517,8 @@
 
 	ret = component_add(dev, &dw_mipi_dsi_imx_ops);
 	if (ret) {
-		DRM_DEV_ERROR(dev, "failed to register component: %d\n", ret);
+		if (ret != -EPROBE_DEFER)
+			DRM_DEV_ERROR(dev, "failed to register component: %d\n", ret);
 		dw_mipi_dsi_remove(dsi->dmd);
 	}
 
