--- a/ssl/statem/statem_srvr.c
+++ b/ssl/statem/statem_srvr.c
@@ -2015,6 +2015,11 @@
          */
         int master_key_length;
 
+        // hostapd eap-fast failure
+        // work-around for missing allocation of s->s3.tmp.valid_flags?
+        if (s->s3.tmp.valid_flags == NULL)
+            s->s3.tmp.valid_flags = OPENSSL_zalloc(s->ssl_pkey_num * sizeof(uint32_t));
+
         master_key_length = sizeof(s->session->master_key);
         if (s->ext.session_secret_cb(ussl, s->session->master_key,
                 &master_key_length, ciphers,
